mods/other/rocket_launcher/init.lua

```diff
@@ -40,7 +39,7 @@ minetest.register_tool("rocket_launcher:launcher", {
 			end
 			minetest.sound_play('fire_extinguish_flame',{to_player = name, gain = 0.5})
 			itemstack:set_wear(WEAR_MAX - 6000)
-			ctf_modebase.update_wear.start_update(user:get_player_name(), itemstack, WEAR_MAX/4, true)
+			ctf_modebase.update_wear.start_update(user:get_player_name(), "rocket_launcher:launcher", WEAR_MAX/4, true)
```

Comment:
I think you shouldn't have changed this. If you want item name of an item stack, you should use the Luanti API and use `itemstack:get_name()`. Your code will work in this case. But not any code which works has been written right.

For instance in this case, if we later wanted to change the item name from `rocket_launcher:launcher` to something else, we'll have to remember to change this line as first. Which means more headache for us. And if we forget, a regression will happen and then we would pull our hair to find out what happened :)))


In software engineering a regression happens when you make changes, like add a new feature, and then a bug or a problem will appear as a side effect of our change. We should write our software in the way to prevent regressions from happening.